### ISSUE TITLE
Intermittant CI failure tracking issue II

### ISSUE THREAD

**Issue by elsirion** [fedimint/fedimint#2825]
Successor of #222.

https://github.com/fedimint/fedimint/actions/runs/5637590680/job/15270864510?pr=2824

**Comment by justinmoon** [comment]
[cli_test_rust_tests_bitcoind](https://github.com/fedimint/fedimint/actions/runs/5637590680/job/15272272877?pr=2824#step:9:8891)

**Comment by justinmoon** [comment]
[net::peers::tests::test_connect](https://github.com/fedimint/fedimint/actions/runs/5694166120/job/15436697329?pr=2854#step:6:4063)

**Comment by elsirion** [comment]
* [Shutdown of servers not working in coverage](https://github.com/fedimint/fedimint/actions/runs/5683447905/job/15430786735)
* [`net::peers::tests::test_connect`](https://github.com/fedimint/fedimint/actions/runs/5683447905/job/15408416436)

**Comment by elsirion** [comment]
[`cannot_replay_transactions`](https://github.com/fedimint/fedimint/actions/runs/5717261430/job/15490630260) gets tuck

**Comment by justinmoon** [comment]
[net::peers::tests::test_connect](https://github.com/fedimint/fedimint/actions/runs/5717194030/job/15490426718#step:6:3775)

**Comment by justinmoon** [comment]
[config::api::tests::test_config_api](https://github.com/fedimint/fedimint/actions/runs/5716097867/job/15486837776?pr=2419#step:6:4319)

**Comment by justinmoon** [comment]
code coverage timeout https://github.com/fedimint/fedimint/actions/runs/5725354013/job/15514671474?pr=2875

**Comment by justinmoon** [comment]
[net::peers::tests::test_connect](https://github.com/fedimint/fedimint/actions/runs/5717194030/job/15490426718)

**Comment by justinmoon** [comment]
[cli_test_reconnect](https://github.com/fedimint/fedimint/actions/runs/5719826045/job/15498532948?pr=2796)

**Comment by justinmoon** [comment]
[net::peers::tests::test_connect](https://github.com/fedimint/fedimint/actions/runs/5719826045/job/15518195481#step:6:4113)

**Comment by justinmoon** [comment]
[test_gateway_client_pay_invalid_invoice ](https://github.com/fedimint/fedimint/actions/runs/5721960641/job/15504325112?pr=2873#step:9:7133)

**Comment by okjodom** [comment]
[test_gateway_cannot_pay_expired_invoice](https://github.com/fedimint/fedimint/actions/runs/5730061410/job/15528031459)

**Comment by justinmoon** [comment]
[on_chain_peg_in_and_peg_out] (https://github.com/fedimint/fedimint/actions/runs/5746660192/job/15576476256#step:9:792)

**Comment by justinmoon** [comment]
Some debugging ideas from call:

- Compile another project in parallel to introduce CPU contention
- Put a loop around where it fails
- Write bash scripts to re-run one test many times to try to trigger failure

**Comment by justinmoon** [comment]
* [test_gateway_cannot_pay_expired_invoice](https://github.com/fedimint/fedimint/actions/runs/5777285019/job/15657290448#step:9:6985)
* [test_gateway_cannot_pay_expired_invoice](https://github.com/fedimint/fedimint/actions/runs/5777246800/job/15657206103#step:9:6253)
* [cli_load_test_tool_test](https://github.com/fedimint/fedimint/actions/runs/5777213045/job/15657131421#step:9:18306)

**Comment by justinmoon** [comment]
`test_gateway_cannot_pay_expired_invoice` & `test_gateway_client_pay_unpayable_invoice` https://github.com/fedimint/fedimint/actions/runs/5789072035/job/15689333467?pr=2911#step:9:7236

**Comment by elsirion** [comment]
https://github.com/fedimint/fedimint/actions/runs/5888250155/job/15969079807

**Comment by elsirion** [comment]
`runs_consensus_if_tx_submitted` https://github.com/fedimint/fedimint/actions/runs/5900523840/job/16004933342?pr=2976

**Comment by dpc** [comment]
Note: #2980 should make it much easier to debug failed runs.

**Comment by elsirion** [comment]
[Something got all the code coverage tests stuck here](https://github.com/fedimint/fedimint/actions/runs/5984042547/job/16234829285) (search for `has been`).

**Comment by elsirion** [comment]
[All the `backed_test`s failed](https://github.com/fedimint/fedimint/actions/runs/6006343654/job/16290712447?pr=3046#step:9:558)

**Comment by elsirion** [comment]
Flakiness intensified, [a lot of the code coverage tests just get stuck apparently](https://github.com/fedimint/fedimint/actions/runs/6076120840/job/16483599912?pr=3054#step:6:3873).

**Comment by bradleystachurski** [comment]
[`ln-gateway::gatewayd-integration-tests test_gateway_configuration`](https://github.com/fedimint/fedimint/actions/runs/6746291413/job/18339906141?pr=3512)

**Comment by justinmoon** [comment]
https://github.com/fedimint/fedimint/actions/runs/6788652621/job/18461865546?pr=3547 had `peg_out_fail_refund` run too long

**Comment by bradleystachurski** [comment]
```
FAIL [  36.905s] fedimint-ln-gateway::gatewayd-integration-tests test_gateway_executes_swaps_between_connected_federations
```

https://github.com/fedimint/fedimint/actions/runs/7013415232/job/19079522681?pr=3746

**Comment by okjodom** [comment]
> fedimint-workspace-lcov-ci>         SLOW [>420.000s] fedimint-server config::api::tests::test_config_api[1m0 skipped     
fedimint-workspace-lcov-ci>  TERMINATING [>480.000s] fedimint-server config::api::tests::test_config_api[1m0 skipped     
fedimint-workspace-lcov-ci>      TIMEOUT [ 480.026s] fedimint-server config::api::tests::test_config_api skipped     
fedimint-workspace-lcov-ci> --- STDOUT:              fedimint-server config::api::tests::test_config_api ---
fedimint-workspace-lcov-ci> running 1 test
fedimint-workspace-lcov-ci> test config::api::tests::test_config_api has been running for over 60 seconds
fedimint-workspace-lcov-ci> 
fedimint-workspace-lcov-ci> --- STDERR:              fedimint-server config::api::tests::test_config_api ---
fedimint-workspace-lcov-ci> 2024-01-22T19:41:15.341974Z  INFO consensus: Starting config gen
fedimint-workspace-lcov-ci> 2024-01-22T19:41:15.342039Z  INFO net::peer::dkg: Created new config gen Api
fedimint-workspace-lcov-ci> 2024-01-22T19:41:15.343480Z  INFO net::api: Starting api on ws://127.0.0.1:25031
fedimint-workspace-lcov-ci> 2024-01-22T19:41:15.343510Z  INFO consensus: Starting config gen
fedimint-workspace-lcov-ci> 2024-01-22T19:41:15.343601Z  INFO net::peer::dkg: Created new config gen Api
fedimint-workspace-lcov-ci> 2024-01-22T19:41:15.343872Z  INFO net::api: Starting api on ws://127.0.0.1:25033
fedimint-workspace-lcov-ci> 2024-01-22T19:41:15.343892Z  INFO consensus: Starting config gen
fedimint-workspace-lcov-ci> 2024-01-22T19:41:15.343925Z  INFO net::peer::dkg: Created new config gen Api
fedimint-workspace-lcov-ci> 2024-01-22T19:41:16.342903Z  INFO test: Test retrying server status
fedimint-workspace-lcov-ci> 2024-01-22T19:41:16.345235Z  INFO net::peer::dkg: Set password for config gen
fedimint-workspace-lcov-ci> 2024-01-22T19:41:16.345542Z  WARN net::api: api request error path="set_password" error=ApiError { code: 400, message: "Expected to be in AwaitingPassword state" }
fedimint-workspace-lcov-ci> thread 'tokio-runtime-worker' panicked at fedimint-server/src/lib.rs:206:14:
fedimint-workspace-lcov-ci> Could not build API server: API name: config-gen
[2335](https://github.com/fedimint/fedimint/actions/runs/7616433363/job/20743174870#step:6:2336)
fedimint-workspace-lcov-ci> Caused by:
fedimint-workspace-lcov-ci>     0: Bind address: 127.0.0.1:25035
fedimint-workspace-lcov-ci>     1: Networking or low-level protocol error: Address already in use (os error 98)
fedimint-workspace-lcov-ci>     2: Address already in use (os error 98)

https://github.com/fedimint/fedimint/actions/runs/7616433363/job/20743174870#step:6:2315

**Comment by okjodom** [comment]
ci `reconnect_test`  [fails on merge queue](https://github.com/fedimint/fedimint/actions/runs/7644412053/job/20828675127#step:9:152)

**Comment by bradleystachurski** [comment]
```
FAIL [   5.062s] fedimint-ln-gateway::gatewayd-integration-tests test_gateway_register_with_federation
```

https://github.com/fedimint/fedimint/actions/runs/7703671558/job/20994518252?pr=4166#step:6:2124

Timeouts in separate PR (merge queue)

```
Summary [ 505.141s] 200 tests run: 198 passed (2 slow), 2 timed out, 0 skipped
TIMEOUT [ 480.039s] fedimint-mint-tests::fedimint_mint_tests sends_ecash_out_of_band_cancel
TIMEOUT [ 480.027s] fedimint-server config::api::tests::test_restart_setup
```

https://github.com/fedimint/fedimint/actions/runs/7702842445/job/20991977876#step:6:38066

**Comment by bradleystachurski** [comment]
```
FAIL [   9.208s] fedimint-ln-gateway::gatewayd-integration-tests test_gateway_configuration
```

https://github.com/fedimint/fedimint/actions/runs/7704054812/job/20995614440?pr=4169#step:6:1835

**Comment by elsirion** [comment]
We now generally create separate issues for flakes.
